### PR TITLE
Croptype eval task: Maize

### PR DESCRIPTION
**Pull request by gabrieltseng**
- [x] Add a maize dataloader
- [x] Add a maize eval task
- [x] Add it all to `train.py`. ~~Since #41 makes a few changes to `train.py`, I'd like to merge that in before making these final changes.~~

In these results, I report the results of Presto when finetuned on Maize ("maize, full") and when finetuned on crop vs. non crop ("Crop, full").

| Model | Head | F1 | Precision | Recall |
| --- | --- | --- | --- | --- |
| Maize, full | Finetuned | 0.8081 | 0.8635 | 0.7593 |
| Maize, full | Random Forest | 0.8055 | 0.88 | 0.7427 |
| Maize, full | Linear Regression | 0.7144 | 0.5860 | 0.9148 | 
| Maize, full | CatBoost | 0.7684 | 0.6789 | 0.8851 |
| Crop, full | Random Forest | 0.6712 | 0.8725 | 0.5452 |
| Crop, full | Linear Regression | 0.5463 | 0.3950 | 0.8855 | 
| Crop, full | CatBoost | 0.6930 | 0.5741 | 0.8738 |

### CODE REVIEW

File: presto/dataset.py [hunk 1]
```diff
@@ -219,6 +229,13 @@ def join_with_world_df(dataframe: pd.DataFrame) -> pd.DataFrame:
         return joined.to_crs("EPSG:4326")
 
 
+class WorldCerealLabelledMaizeDataset(WorldCerealLabelledDataset):
+    @staticmethod
+    def target_from_row_dict(row_d) -> int:
```
Comment (kvantricht) [inline presto/dataset.py:234]: To be more closely aligned with the hierarchic nature of WorldCereal (first map out crops, then map the crops inside cropland), we should think about how to do that.

in principle `row_d["LANDCOVER_LABEL"] != 11` should indicate which rows are not to be used to train the crop type models. Or maybe  `row_d["LANDCOVER_LABEL"] not in [10, 11, 12, 13]` to also have perennials and grassland in to be sure. I guess Presto can then also focus entirely on the crops themselves?

Is this filtering mechanism something we should do when initializing the dataset with a dataframe?

File: presto/eval.py [hunk 2]
```diff
@@ -50,12 +51,19 @@ def __init__(
         self,
         train_data: pd.DataFrame,
         val_data: pd.DataFrame,
+        predict_maize: bool = False,
```
Comment (kvantricht) [inline presto/eval.py:54]: It seems this can quickly become chaotic. What if we want to map another crop?
Couldn't we work with a list specifying the target label(s) so we support any kind of crop we want?

If we do something like `predict_crops: List = [1200]`, that could be a start (though default value being a list is not a good idea if I recall well. But you get the point, would it make sense?

File: presto/dataset.py (hunk 1 above)
Comment (gabrieltseng) [reply presto/dataset.py]: Added to the eval task

File: presto/eval.py (hunk 2 above)
Comment (gabrieltseng) [reply presto/eval.py]: Okay this makes sense to me. Added a `target_function` option to the eval task (and the dataset) instead of creating a new class.

File: presto/dataset.py [hunk 3]
```diff
@@ -163,6 +174,17 @@ def __getitem__(self, idx):
         )
 
 
+def filter_remove_noncrops(df: pd.DataFrame) -> pd.DataFrame:
+    noncrop_labels = [10, 11, 12, 13]
```
Comment (kvantricht) [inline presto/dataset.py:178]: I'm confused. `[10, 11, 12, 13]` should be considered `crop_labels`, and in a croptype task, we should filter the `df` to only keep the rows which have a `LANDCOVER_LABEL` matching values in this list. It seems the opposite is happening here and we're throwing away all crops? I'd think actually all maize labels would be gone now. Or am I misreading this part?

**Review by kvantricht — CHANGES_REQUESTED** [review]
Awesome generalization of the class, very useful! Just one comment added because I think the filter function now does the opposite of what we want it to do.

File: presto/dataset.py (hunk 3 above)
Comment (gabrieltseng) [reply presto/dataset.py:178]: 🤦 Thanks for catching this

**Review by kvantricht — APPROVED** [review]
